cranelift-codegen/src/legalizer/mod.rs

```diff
@@ -153,8 +165,6 @@ pub fn legalize_function(func: &mut ir::Function, cfg: &mut ControlFlowGraph, is
     // Process EBBs in layout order. Some legalization actions may split the current EBB or append
     // new ones to the end. We need to make sure we visit those new EBBs too.
     while let Some(ebb) = pos.next_ebb() {
-        split::split_ebb_params(pos.func, cfg, ebb);
-
```

Comment:
Actually, now that I look at this I realize that I don't exactly understand what this is solving: why does splitting everything before this loop allow your test to compile?

Reply:
Or in other words, why do we have to split above (with a separate `FuncCursor`) and not here?

Reply:
Taking the test as an example

Before when `ebb4` would get visited, `split_ebb_params` would insert an `isplit` before every jump to `ebb4`, as the jump in `ebb2` is before `ebb4`, the new `isplit` would never get legalized. After this PR, first all ebb params get split, introducing the necessary `isplit` and then all instructions get legalized, including the new `isplit`.

Reply:
Ah, makes sense.